packages/framework/src/hooks/useStyleNames.ts

```diff
@@ -1,28 +1,66 @@
 import { useAtomValue } from "jotai";
-import type { CSSProperties } from "react";
-import { type Expression, resolveStyleNames } from "../shared";
+import { CSSProperties, useMemo } from "react";
+import { get, intersection, isEmpty, isString, keys, merge } from "lodash-es";
+import { type EnsembleThemeModel, type Expression } from "../shared";
 import { themeAtom } from "../state";
-import { isString } from "lodash-es";
+import { useEvaluate } from "./useEvaluate";
+
+const resolveStyleNames = (
+  styleNames: string,
+  theme: EnsembleThemeModel,
+): CSSProperties => {
+  const appStyleNames = keys(theme.Styles);
+  const overlappingNames = intersection(styleNames.split(" "), appStyleNames);
+  if (isEmpty(overlappingNames)) {
+    return {};
+  }
+
+  const appliedStyles = {};
+  overlappingNames.forEach((name) => {
+    merge(appliedStyles, get(theme.Styles, name));
+  });
+
+  return appliedStyles;
+};
 
 export const useStyleNames = (
   namedStyles?: Expression<string | string[]>,
   classStyles?: Expression<string | string[]>,
 ): CSSProperties | undefined => {
   const themeContext = useAtomValue(themeAtom);
 
-  let dotClassStyles = "";
-  if (isString(classStyles) && classStyles.length > 0) {
-    dotClassStyles = classStyles
-      .trim()
-      .split(" ")
-      .map((className) => `.${className}`)
-      .join(" ");
-  }
-  const styleNames = `${
-    isString(namedStyles) ? namedStyles : ""
-  } ${dotClassStyles}`;
+  // evaluate styles
+  const nameStylesArray: string[] | undefined = isString(namedStyles)
+    ? // eslint-disable-next-line prefer-named-capture-group
+      namedStyles.split(/([a-zA-Z0-9_-]+)\s+(\${.+?})\s+(\w+)/).filter((x) => x)
+    : namedStyles;
 
-  if (styleNames && themeContext) {
-    return resolveStyleNames(styleNames, themeContext);
-  }
+  // evaluate classnames
+  const classStylesArray: string[] | undefined = isString(classStyles)
+    ? // eslint-disable-next-line prefer-named-capture-group
+      classStyles.split(/([a-zA-Z0-9_-]+)\s+(\${.+?})\s+(\w+)/)
```

Comment:
Can we reuse the same regex everywhere, i.e. from fetcher.ts?

Reply:
No, because this regex is for split string, so we can convert string into array of classname or stylename, and that classname or stylename, may be expression or not.

and the regex added in fetcher is for checking that given string is expression or not.